packages/api/src/record.ts

```diff
@@ -151,26 +150,39 @@ export class Record implements RecordModel {
   constructor(agent: Web5Agent, options: RecordOptions) {
     this._agent = agent;
 
-    // Store the target and author DIDs that were used to create the message to use for subsequent reads, etc.
+    /** Store the target and author DIDs that were used to create the message to use for subsequent
+     * updates, reads, etc. */
     this.author = options.author;
     this.target = options.target;
 
+    /** If the record was queried from a remote DWN, the `remoteTarget` DID will be defined. This
+     * value is used to send subsequent read requests to the same remote DWN in the event the
+     * record's data payload was too large to be returned in query results. */
+    this._remoteTarget = options.remoteTarget;
+
     // RecordsWriteMessage properties.
     this._attestation = options.attestation;
     this._contextId = options.contextId;
     this._descriptor = options.descriptor;
     this._encryption = options.encryption;
     this._recordId = options.recordId;
 
+    if (options.encodedData) {
+      // If `encodedData` is set, then it is expected that:
+      // type is Blob if the Record object was instantiated by dwn.records.create()/write().
+      // type is Base64 URL encoded string if the Record object was instantiated by dwn.records.query().
+      // If it is a string, we need to Base64 URL decode to bytes and instantiate a Blob.
+      this._encodedData = (typeof options.encodedData === 'string') ?
+        new Blob([Convert.base64Url(options.encodedData).toUint8Array()], { type: this.dataFormat }) :
```

Comment:
I've noticed that in DWN and affiliated libs we use `base64url` rather than `base64` (atob/btoa). Is there a good reason for this?

Reply:
@shamilovtim Base64 URL encoding is the format used in a variety of other related standards (e.g., JWK, DIDs, VCs, etc.).

Chat GTP answer:
The key difference between base64url and regular base64 encoding lies in the characters used for encoding. In regular base64 encoding (as used by atob and btoa in JavaScript), the characters + and / are used. However, these characters can cause issues in URLs and filenames since they have special meanings in those contexts.

base64url encoding addresses this by replacing + with - and / with _. This slight modification makes it URL and filename safe. The padding character = is also typically omitted in base64url to further enhance URL-friendliness.

Reply:
That's where I'm confused. I assume the DWN data is never used in a URL as its intended for file / data transfer. So should we consider the disadvantages of using [pure JS implementations of base64](https://github.com/multiformats/js-multiformats/blob/master/src/bases/base.js#L291C30-L291C30) over browser native ([atob/btoa implemented in C++](https://chromium.googlesource.com/external/github.com/flutter/engine/+/demo_apk_23/sky/engine/core/frame/DOMWindowBase64.cpp)) implementations of base64?